crates/core/component/ibc/src/ics20_withdrawal.rs

```diff
@@ -84,17 +80,18 @@ impl EffectingData for Ics20Withdrawal {
             .personal(b"PAH:ics20wthdrwl")
             .to_state();
 
-        let destination_chain_id_hash =
-            blake2b_simd::Params::default().hash(self.destination_chain_id.as_bytes());
         let destination_chain_address_hash =
             blake2b_simd::Params::default().hash(self.destination_chain_address.as_bytes());
+        let return_address = blake2b_simd::Params::default().hash(&self.return_address.to_vec());
 
-        state.update(destination_chain_id_hash.as_bytes());
         state.update(&self.value().amount.to_le_bytes());
         state.update(&self.value().asset_id.to_bytes());
+        state.update(&self.denom.encode_to_vec());
```

Comment:
i don't think we actually need this, since asset_id is computed by denom and covered by the hash, but I included it here for completeness